test/asyncio_tests/test_asyncio_change_stream.py

```diff
@@ -260,3 +263,24 @@ async def test_database(self):
             i += 1
             if i == 3:
                 break
+
+    @asyncio_test(timeout=10)
+    async def test_iterate_more_streams_than_workers(self):
+        # Create more tasks running ChangeStream.next than there are worker
+        # threads, and then ensure that other tasks can still run.
+        streams = [self.collection.watch() for _ in range(max_workers)]
+        tasks = [stream.next() for stream in streams]
+        try:
+            async def find_insert():
+                # Wait for all change streams to be created
+                while not all(stream.delegate for stream in streams):
+                    await asyncio.sleep(.1)
+                await self.collection.find_one()
+                await self.collection.insert_one({})
+            tasks.extend([find_insert() for _ in range(10)])
+            await asyncio.gather(*tasks)
```

Comment:
I would expect this command to raise an exception that will cause the test fail. Why doesn't it?